docs/content/modules/products/serverless-module.md

```diff
@@ -0,0 +1,382 @@
+    },
+  ]
+  
+  const productsData = [
+    {
+      id: "test-1",
+      title: "product 1",
+      status: "published",
+      descriptions: "Lorem ipsum dolor sit amet, consectetur.",
+      tags: [
+        {
+          id: "tag-1",
+          value: "France",
+        },
+      ],
+      categories: [
+        {
+          id: "category-0",
+        },
+      ],
+    },
+    {
+      id: "test-2",
+      title: "product",
+      status: "published",
+      descriptions: "Lorem ipsum dolor sit amet, consectetur.",
+      tags: [
+        {
+          id: "tag-2",
+          value: "Germany",
+        },
+      ],
+      categories: [
+        {
+          id: "category-1",
+        },
+      ],
+    },
+  ]
+  
+  const variantsData = [
+    {
+      id: "test-1",
+      title: "variant title",
+      sku: "sku 1",
+      product: { id: productsData[0].id },
+      inventory_quantity: 10,
+    },
+    {
+      id: "test-2",
+      title: "variant title",
+      sku: "sku 2",
+      product: { id: productsData[1].id },
+      inventory_quantity: 10,
+    },
+  ]
+  
+  module.exports = {
+    productCategoriesData,
+    productsData,
+    variantsData,
+  }
+  ```
+
+</details>
+
+Then run the first and optionally second commands to migrate the database schema:
+
+```bash npm2yarn
+npm run product:migrations:run
+# optionally
+npm run product:seed
+```
+
+### Step 4: Adjust Next.js Configurations
+
+Next.js uses Webpack for compilation. Since quite a few of the dependencies used by the product module are not Webpack optimized, you have to add the product module as an external dependency.
+
+To do that, add the `serverComponentsExternalPackages` option in `next.config.js`:
+
+```js title=next.config.js
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  experimental: {
+    serverComponentsExternalPackages: [
+      "@medusajs/product",
+    ],
+  },
+}
+
+module.exports = nextConfig
+```
+
+### Step 5: Create API Route
+
+The product module is ready for use now! You can now use it to create API endpoints within your Next.js application.
+
+For example, create the file `app/api/products/route.ts` with the following content:
+
+```ts title=app/api/products/route.ts
+import { NextResponse } from "next/server"
+
+import { 
+  initialize as initializeProductModule,
+} from "@medusajs/product"
+
+export async function GET(request: Request) {
+  const productService = await initializeProductModule()
+
+  const data = await productService.list({})
+
+  return NextResponse.json({ products: data })
+}
+```
+
+This creates a `GET` endpoint at the route `/api/products`. You import the `initialize` function, aliased as `initializeProductModule`, from `@medusajs/product`. Then, in the endpoint, you invoke the `initializeProductModule` function, which returns an instance of the `ProductService`. Using the product service’s `list` method, you retrieve all available products and return them in the response of the endpoint.
```

Comment:
Incase they go into the code, the correct instance name would be `ProductModuleService`